test/integration/src/test/java/io/pravega/test/integration/ReadFromDeletedStreamTest.java

```diff
@@ -54,11 +54,12 @@ public void testDeletedAndRecreatedStream() throws Exception {
         @Cleanup
         MockClientFactory clientFactory = streamManager.getClientFactory();
         // Mocking pravega service by putting scale up and scale down requests for the stream
-        @Cleanup
```

Comment:
Why no cleanup anymore? If your test fails prior to reaching line 63, this will stay open and won't release any resources it acquired.

Reply:
close() is invoked in the next line.